// @prisma/client/wasm.js

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal,
  objectEnumValues,
  makeStrictEnum,
  Public,
  getRuntime,
  skip
} = require('./runtime/index-browser.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 6.5.0
 * Query Engine version: 173f8d54f8d52e692c7e27e72a88314ec7aeff60
 */
Prisma.prismaVersion = {
  client: "6.5.0",
  engine: "173f8d54f8d52e692c7e27e72a88314ec7aeff60"
}

Prisma.PrismaClientKnownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientKnownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientUnknownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientRustPanicError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientRustPanicError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientInitializationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientInitializationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientValidationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientValidationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`sqltag is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.empty = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`empty is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.join = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`join is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.raw = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`raw is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.getExtensionContext is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.defineExtension = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.defineExtension is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}



/**
 * Enums
 */

exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.Ep_clientsScalarFieldEnum = {
  cli_id: 'cli_id',
  cli_name: 'cli_name',
  cli_lastname: 'cli_lastname',
  cli_ced: 'cli_ced',
  cli_email: 'cli_email',
  cli_version: 'cli_version'
};

exports.Prisma.Ep_informartionScalarFieldEnum = {
  info_id: 'info_id',
  info_name: 'info_name',
  info_location: 'info_location',
  info_spaces: 'info_spaces',
  info_version: 'info_version',
  info_image: 'info_image',
  info_owner: 'info_owner',
  info_owner_ced: 'info_owner_ced',
  info_owner_phone: 'info_owner_phone',
  info_schedule: 'info_schedule'
};

exports.Prisma.Ep_packagesScalarFieldEnum = {
  pack_id: 'pack_id',
  pack_name: 'pack_name',
  pack_price: 'pack_price',
  pack_version: 'pack_version'
};

exports.Prisma.Ep_roleScalarFieldEnum = {
  rol_id: 'rol_id',
  rol_name: 'rol_name',
  rol_version: 'rol_version'
};

exports.Prisma.Ep_serverScalarFieldEnum = {
  server_id: 'server_id',
  server_type: 'server_type',
  server_port: 'server_port',
  server_protocol: 'server_protocol',
  server_mail: 'server_mail',
  server_password: 'server_password',
  server_timeout: 'server_timeout',
  server_version: 'server_version'
};

exports.Prisma.Ep_taxScalarFieldEnum = {
  tax_id: 'tax_id',
  tax_price: 'tax_price',
  tax_version: 'tax_version'
};

exports.Prisma.Ep_usersScalarFieldEnum = {
  users_id: 'users_id',
  users_name: 'users_name',
  users_lastname: 'users_lastname',
  users_ced: 'users_ced',
  users_version: 'users_version',
  user_username: 'user_username',
  user_password: 'user_password',
  users_email: 'users_email'
};

exports.Prisma.Ep_vehiclesScalarFieldEnum = {
  veh_id: 'veh_id',
  veh_plate: 'veh_plate',
  veh_reference: 'veh_reference',
  veh_status: 'veh_status',
  veh_owner: 'veh_owner',
  veh_color: 'veh_color',
  veh_ingress_date: 'veh_ingress_date',
  veh_egress_date: 'veh_egress_date',
  veh_tax: 'veh_tax',
  veh_version: 'veh_version'
};

exports.Prisma.Ep_cash_per_dayScalarFieldEnum = {
  cpd_id: 'cpd_id',
  cpd_amount: 'cpd_amount',
  cpd_user_id: 'cpd_user_id',
  cpd_date: 'cpd_date',
  cpd_version: 'cpd_version'
};

exports.Prisma.Ep_cashierScalarFieldEnum = {
  cash_id: 'cash_id',
  cash_type: 'cash_type',
  cash_amount: 'cash_amount',
  cash_date: 'cash_date',
  cash_version: 'cash_version',
  cash_movement_type: 'cash_movement_type'
};

exports.Prisma.Ep_cashier_transactionScalarFieldEnum = {
  cash_trans_id: 'cash_trans_id',
  cash_trans_amount: 'cash_trans_amount',
  cash_trans_type: 'cash_trans_type',
  cash_trans_method: 'cash_trans_method',
  cash_trans_date: 'cash_trans_date',
  cash_trans_cashier_id: 'cash_trans_cashier_id'
};

exports.Prisma.Ep_notesScalarFieldEnum = {
  ep_note_id: 'ep_note_id',
  ep_note_user: 'ep_note_user',
  ep_note_content: 'ep_note_content',
  ep_note_date: 'ep_note_date'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};


exports.Prisma.ModelName = {
  ep_clients: 'ep_clients',
  ep_informartion: 'ep_informartion',
  ep_packages: 'ep_packages',
  ep_role: 'ep_role',
  ep_server: 'ep_server',
  ep_tax: 'ep_tax',
  ep_users: 'ep_users',
  ep_vehicles: 'ep_vehicles',
  ep_cash_per_day: 'ep_cash_per_day',
  ep_cashier: 'ep_cashier',
  ep_cashier_transaction: 'ep_cashier_transaction',
  ep_notes: 'ep_notes'
};

/**
 * This is a stub Prisma Client that will error at runtime if called.
 */
class PrismaClient {
  constructor() {
    return new Proxy(this, {
      get(target, prop) {
        let message
        const runtime = getRuntime()
        if (runtime.isEdge) {
          message = `PrismaClient is not configured to run in ${runtime.prettyName}. In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters
`;
        } else {
          message = 'PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `' + runtime.prettyName + '`).'
        }
        
        message += `
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report`

        throw new Error(message)
      }
    })
  }
}

exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
